L60-core-remove-custom-allocator.md

```diff
@@ -0,0 +1,92 @@
+Remove custom allocation function overrides from core surface API
+----
+* Author(s): veblush
+* Approver: mark
+* Status: Draft
+* Implemented in: https://github.com/grpc/grpc/pull/20462
+* Last updated: 2019-10-09
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/kDswzg-cEZU
+
+## Abstract
+
+Remove `gpr_get_allocation_functions` and `gpr_set_allocation_functions`
+from the core surface API.
+
+## Background
+
+Since gRPC Core was allowed to use the C++ standard library, it's getting
+harder to keep using the gRPC allocator which can be overriden by users.
+Previously, it was fairly straightforward to use a family of gpr memory
+allocation functions such as `gpr_malloc` and `gpr_free` because most of
+the code was written in C language.
+Once C++ started being used heavily, however, using it makes code harder to
+read and sometimes impossible to do it.
+Instead of maintaining these functions, this proposes to remove them.
+
+### Related Proposals:
+
+N/A
+
+## Proposal
+
+Following functions managing custom memory allocator will be removed.
+
+- `gpr_get_allocation_functions`
+- `gpr_set_allocation_functions`
+
+gRPC memory allocation functions such as `gpr_malloc` and `gpr_free` will
+remain with this change because they have been used when data allocated in
```

Comment:
If this is the only remaining use-case for these functions, does that mean that we don't need to use these functions inside of core?  In other words, external code that creates metadata to pass it into us will need to use these, but once inside of core, we can use `free()` directly instead of `gpr_free()`.  Is that right?

Reply:
That's right. Since they're identical inside, it's up to us deciding which gRPC core will use.